packages/ckeditor5-typing/tests/utils/injectunsafekeystrokeshandling.js

```diff
@@ -129,5 +130,30 @@ describe( 'unsafe keystroke handling utils', () => {
 				return editor.model.change( writer => writer.batch );
 			}
 		} );
+
+		describe( 'handling Shift + Delete on Windows', () => {
```

Comment:
Maybe tests on other platforms should be here also. Seems like all the new cases tested in `DeleteObserver` would be needed here.

Reply:
Added one test for non-Windows platform and also 2 other to check if new condition does not interfere when it should not.

It is not possible to add all new test cases that are already added for `DeleteObserver`, but only the ones that operate on a non-collapsed selection. The tests in this file uses only the `Typing` plugin (as expected) and the `injectUnsafeKeystrokesHandling()` function deletes only the selected content.